src/api/analytics.ts

```diff
@@ -0,0 +1,36 @@
+import type { GatsbyFunctionRequest, GatsbyFunctionResponse } from 'gatsby'
+
+import sendGAEvent from '../server/GoogleAnalytics'
+
+const validateRequest = (
+  req: GatsbyFunctionRequest,
+  res: GatsbyFunctionResponse
+) => {
+  if (req.method !== 'POST') {
+    res.statusCode = 405
+    res.send('')
+
+    return false
+  }
+
+  return true
+}
+
+const handler = async (
```

Comment:
I don't know if it should happen here, but I feel we should also use CORS blocking for requests coming from domains other than the store.